src/test/librados/watch_notify.cc

```diff
@@ -283,6 +283,7 @@ TEST_F(LibRadosWatchNotify, Watch2Delete) {
   ASSERT_EQ(-ENOTCONN, notify_err);
   ASSERT_EQ(-ENOTCONN, rados_watch_check(ioctx, handle));
   rados_unwatch2(ioctx, handle);
+  rados_watch_flush(cluster);
```

Comment:
this looks good to me. probably we can do the cleanup in `TearDown()`, as the fixture is designed for this purpose. like

```diff
diff --git a/src/test/librados/watch_notify.cc b/src/test/librados/watch_notify.cc
index e28fd52711..e408b90abf 100644
--- a/src/test/librados/watch_notify.cc
+++ b/src/test/librados/watch_notify.cc
@@ -55,6 +55,10 @@ protected:
                                     void *data,
                                     size_t data_len);
   static void watch_notify2_test_errcb(void *arg, uint64_t cookie, int err);
+  void TearDown() override {
+    RadosTest::TearDown();
+    rados_watch_flush(cluster);
+  }
 };


@@ -101,6 +105,10 @@ protected:
   int notify_err = 0;

   friend class WatchNotifyTestCtx2;
+  void TearDown() override {
+    RadosTestParamPP::TearDown();
+    cluster.watch_flush();
+  }
 };

 IoCtx *notify_ioctx;
@@ -831,6 +839,8 @@ TEST_F(LibRadosWatchNotify, Watch3Timeout) {

   // re-watch
   rados_unwatch2(ioctx, handle);
+  rados_watch_flush(cluster);
+
   handle = 0;
   ASSERT_EQ(0,
            rados_watch2(ioctx, notify_oid, &handle,
```